### ISSUE TITLE
A few more ruby adjustments

### ISSUE THREAD

**Issue by sandstrom** [standard-webhooks/standard-webhooks#194]
Since my previous issue #158 got merged, I would like to suggest a few more adjustments.

Mostly to make the code base better with how code is written in the ruby community, but some are also functional.

Let me know which of the below you think is a good idea, and I'll open a PR.

### Suggestions

- [ ] Proper private constants (https://blog.toshima.ru/2020/02/22/ruby-private-constant.html)
- [ ] Adding a minimum ruby version to the gemspec (would you prefer 3.1 or 3.0?; 3.0 is already end of life, but still possible to support)
- [ ] Remove `symbolize_names` from JSON.parse.
    
    These aren't symbols according to at least this definition:
    
    > “If the contents (the sequence of characters) of the object are important, use a string. If the identity of the object is important, use a symbol.”
- [ ] Remove explicit return

### Questions

1. There is both `secure_compare` and `fixed_with_secure_compare` in the Ruby OpenSSL library[1]. The difference is that the former will hash to ensure fixed length. But if both have fixed length, then there is no need.

What's the case here? Are we certain the two are fixed and identical length already? Otherwise we should consider switching to `secure_compare`.

(note that this is already what the 'fallback' implementation guards against: `raise ArgumentError, "string length mismatch." unless a.bytesize == b.bytesize`)

2. Are there any features lacking in the Ruby sdk, that you have in the other SDKs and want me to look at?

[1] https://github.com/ruby/openssl/pull/280

cc @tasn @zekth

**Comment by zekth** [comment]
Related to you questions and points i think everything has been implemented in https://github.com/standard-webhooks/standard-webhooks/pull/192/files am i right? I think we might be able to close this issue. Could you confirm?

**Comment by sandstrom** [comment]
No, this is a new issue that I opened after merging the first one. 

I wanted to break up the work in smaller pieces. 

That said, what I propose here isn't anything major, I'll do it in 15-30 minutes.

**Comment by sandstrom** [comment]
Ping @zekth

**Comment by sandstrom** [comment]
friendly ping @zekth 😄

**Comment by zekth** [comment]
Sorry this got completely buried in the notifications.

> What's the case here? Are we certain the two are fixed and identical length already? Otherwise we should consider switching to secure_compare.

I'm not exactly sure but this uses `rails` code: https://github.com/standard-webhooks/standard-webhooks/blob/a1773d7ffc577d730e2bf4d4bd740a82a42aa904/libraries/ruby/lib/standardwebhooks/util.rb#L5

Your suggestions are legit and PRs are welcome

**Comment by sandstrom** [comment]
Thanks @zekth I've added a PR.
